internal/fsutil/os_detection.go

```diff
@@ -0,0 +1,55 @@
+package fsutil
+
+import (
+	"os"
+	"strings"
+)
+
+type OSInfo struct {
+	Name    string
+	Version string
+}
+
+type Detector interface {
+	HasLinuxFile() bool
+	ReadLinuxFile() (string, error)
+	GetInfo(osReleaseContents string) OSInfo
+}
+
+type Detect struct {
+}
+
+func (d *Detect) HasLinuxFile() bool {
```

Comment:
```suggestion
func (d *Detect) HasSystemdFile() bool {
```
As you've educated me on the finer points here, maybe this will help others understand why we didn't go with the suggestion mentioned [here](https://github.com/buildpacks/lifecycle/pull/1050/files#r1153496828)